sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1142,7 +1142,7 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
             throw QueryCompilationErrors.unsupportedViewOperationError(identifier, cmd, false, u)
```

Comment:
We probably need a new error class for it: `EXPECT_PERSISTENT_VIEW_NOT_TEMP`

Reply:
Done

Reply:
I also added this new error class: `EXPECT_TABLE_OR_PERMANENT_VIEW_NOT_TEMP`.